app/controllers/bpldc/authorities_controller.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module Bpldc
+  class AuthoritiesController < ApplicationController
+    # GET /bpldc/authorities
+    def index
+      @objects = Bpldc::Authority.public_attributes.all
+      render json: results_for_api
+    end
+
+    # GET /bpldc/authorities/subjects
+    def subjects
+      @objects = Bpldc::Authority.public_attributes.subjects
+      render json: results_for_api
+    end
+
+    # GET /bpldc/authorities/subjects
+    def genres
+      @objects = Bpldc::Authority.public_attributes.genres
+      render json: results_for_api
+    end
+
+    # GET /bpldc/authorities/subjects
+    def names
+      @objects = Bpldc::Authority.public_attributes.names
+      render json: results_for_api
+    end
+
+    # GET /bpldc/authorities/subjects
+    def geographics
+      @objects = Bpldc::Authority.public_attributes.geographics
+      render json: results_for_api
+    end
+
+    private
+
+    def results_for_api
```

Comment:
Any particular reason the results for this is different than just just doing `render json: @objects`? Just curious

Reply:
So, when the `public_attributes` scope is used, it limits the attributes to just the ones we need in the API (removes `created_at`, etc.).

But, Rails seems to have a thing where you can't actually remove the `id` attribute, though it is returned as `nil`...
```
[1] pry(main)> Bpldc::Authority.public_attributes.first
=> #<Bpldc::Authority:0x0000563016efd688
 id: nil,
 name: "Art and Architecture Thesaurus",
 code: "aat",
 base_url: "http://vocab.getty.edu/aat">
```

...which then becomes `"id": null` in the JSON output.

So the `results_for_api` is just a way to remove that `"id": null`.

Reply:
One way to omit nil values is adding the oj gem and creating an iniitalizer to set up default values.

In Gemfile add

`gem 'oj', '~> 3.10'`

the create an initializer in 
`config/initializers/oj.rb` with the following

```
  Oj.optimize_rails
      Oj.default_options =
      {
        mode: :rails,
        time_format: :ruby,
        hash_class: ActiveSupport::HashWithIndifferentAccess,
        omit_nil: true
      }
```
The `omit_nil` option will exclude nil values once it is dumped to `json`

Reply:
One thing i forgot to mention is yo should change 
`@objects.map { |rec| rec.as_json.compact }` to  `Oj.dump(@objects.as_json)`